spec/integ/bridge.spec.js

```diff
@@ -187,6 +187,22 @@ describe("Bridge", function() {
             });
         });
 
+        it("should not create a room if roomId is returned from the function but should still store it",
+        async function(done) {
+            bridgeCtrl.onAliasQuery.and.returnValue({ roomId: "!abc123:bar" });
+            await bridge.run(101, {}, appService);
+
+            appService.onAliasQuery("#foo:bar");
+
+            expect(clients["bot"].createRoom).not.toHaveBeenCalled();
+
+            const room = await bridge.getRoomStore().getMatrixRoom("!abc123:bar");
+            expect(room).toBeDefined();
```

Comment:
This bottom part to assert the `room` is just taken from the tests below but I am pretty confused about it testing anything.

If you actually debug this, `room` is `null` and the test seems to support both cases so that is why it passes. Seems to be a problem with all of the tests here.

Reply:
Feels like you need to await `onAliasQuery` to ensure that the promises all run here? This might be a bug with the other tests.

Reply:
With:
```js
            bridgeCtrl.onAliasQuery.and.returnValue({ roomId: "!abc123:bar" });
            await bridge.run(101, {}, appService);

            await appService.onAliasQuery("#foo:bar");

            expect(clients["bot"].createRoom).not.toHaveBeenCalled();

            const room = await bridge.getRoomStore().getMatrixRoom("!abc123:bar");
            expect(room).toBeDefined();
            expect(room.getId()).toEqual("!abc123:bar");
```

it seems to pass :)

Reply:
Thanks for spotting that! Just copy pasting everything from the other tests. We should fix those up too in another PR

Reply:
Created https://github.com/matrix-org/matrix-appservice-bridge/pull/289 to fix the other tests